src/zenml/materializers/pandas_series_materializer.py

```diff
@@ -0,0 +1,78 @@
+#  Copyright (c) ZenML GmbH 2021. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+import os
+import tempfile
+from typing import Any, Type
+
+import pandas as pd
+
+from zenml.artifacts import DataArtifact, SchemaArtifact, StatisticsArtifact
+from zenml.io import fileio
+from zenml.materializers.base_materializer import BaseMaterializer
+
+
+DEFAULT_FILENAME = "df.parquet.gzip"
+COMPRESSION_TYPE = "gzip"
+
+
+class PandasSeriesMaterializer(BaseMaterializer):
+    """Materializer to read data to and from pandas Series."""
+
+    ASSOCIATED_TYPES = (pd.Series,)
+    ASSOCIATED_ARTIFACT_TYPES = (
+        DataArtifact,
+        StatisticsArtifact,
+        SchemaArtifact,
+    )
+
+    def handle_input(self, data_type: Type[Any]) -> pd.Series:
+        """Reads pd.Dataframe from a parquet file and returns as pd.Series."""
+        super().handle_input(data_type)
+        filepath = os.path.join(self.artifact.uri, DEFAULT_FILENAME)
+
+        # Create a temporary folder
+        temp_dir = tempfile.mkdtemp(prefix="zenml-temp-")
+        temp_file = os.path.join(str(temp_dir), DEFAULT_FILENAME)
+
+        # Copy from artifact store to temporary file
+        fileio.copy(filepath, temp_file)
+
+        # Load the model from the temporary file
+        df = pd.read_parquet(temp_file)
+
+        # Cleanup and return
+        fileio.rmtree(temp_dir)
+
+        return df[df.columns[0]]
+
+    def handle_return(self, series: pd.Series) -> None:
```

Comment:
Rather than doing this all again, how about just calling the PandasMaterializer here

```
from zenml.materializers.pandas_materializer import PandasMaterializer

PandasMaterializer(self.artifact).handle_return(series)
```

Reply:
`PandasMaterializer(self.artifact).handle_return` expects `pd.DataFrame` and errors when given `pd.Series`.

Reply:
Perhaps that even a better approach. In the [pandas materializer](https://github.com/zenml-io/zenml/blob/main/src/zenml/materializers/pandas_materializer.py) you can do:

```python
ASSOCIATED_TYPES = (pd.DataFrame, pd.Series, )
```

and then the type hints would be `Union[pd.DataFrame, pd.Series]`. You then just need to handle the logic inside the function:

```
def handle_input(self, data_type: Union[pd.DataFrame, pd.Series]) -> Union[pd.DataFrame, pd.Series]:
    # do normal handle return function
    if data_type is pd.Series:
            # convert to pandas dataframe and return


def handle_return(self, df: Union[pd.DataFrame, pd.Series]) -> None:
      if type(df) is  pd.Series:
              # convert to pandas dataframe
       # do the normal function with dataframes
```

Reply:
I was thinking the same thing.

Reply:
If I convert the series to a df in the materializer handle_return how do I know that I was originally a pd.Series in handle_input?